From e50322645cfc1eac507ea1b44c4e99a156fb983f Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Fri, 26 Jul 2013 10:58:52 +0200
Subject: [PATCH] Free NSS objects in --external-ca scenario

In external CA installation, ipa-server-install leaked NSS objects
which caused an installation crash later when a subsequent call of
NSSConnection tried to free them.

Properly freeing the NSS objects avoid this crash.

https://fedorahosted.org/freeipa/ticket/3773
---
 install/tools/ipa-server-install | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/install/tools/ipa-server-install b/install/tools/ipa-server-install
index 70e5153d7bf88f2ae71407e90dea2de010b97b1c..924285927e6995244f14d5456baeeefc35167a3a 100755
--- a/install/tools/ipa-server-install
+++ b/install/tools/ipa-server-install
@@ -675,18 +675,23 @@ def main():
             sys.exit(1)
 
         certdict = dict((DN(str(cert.subject)), cert) for cert in extchain)
+        del extchain
         certissuer = DN(str(extcert.issuer))
         if certissuer not in certdict:
             print "The PKCS#10 certificate is not signed by the external CA (unknown issuer %s)." % certissuer
             sys.exit(1)
 
         cert = extcert
+        del extcert
         while cert.issuer != cert.subject:
             certissuer = DN(str(cert.issuer))
             if certissuer not in certdict:
                 print "The external CA chain is incomplete (%s is missing from the chain)." % certissuer
                 sys.exit(1)
+            del cert
             cert = certdict[certissuer]
+        del certdict
+        del cert
 
     print "=============================================================================="
     print "This program will set up the FreeIPA Server."
-- 
2.4.3

